templates/js-cypress/main.js

```diff
@@ -0,0 +1,49 @@
+import { Actor } from 'apify';
+import fs from 'fs';
+import cypress from 'cypress';
+import { globby } from 'globby';
+import log from '@apify/log';
+
+await Actor.init();
+
+const input = await Actor.getInput();
+
+const runOneSpec = (spec) => {
+    return cypress.run({
+        config: input,
+        spec,
+    });
+};
+
+log.info(`Running tests with following input: ${JSON.stringify(input)}`);
+
+const tests = await globby('./cypress/e2e/*-spec.cy.js');
+
+log.info(`Getting tests: ${tests}`);
+
+const kvs = await Actor.openKeyValueStore();
+const dataset = await Actor.openDataset();
+for (const test of tests) {
+    const result = await runOneSpec(test);
+    let keyValueStoreLink;
+    if (result?.config?.video) {
+        const baseName = test.split('/').pop();
+        const file = `./cypress/videos/${baseName}.mp4`;
+        const kvsKeyName = baseName.replaceAll('.', '-');
+        await kvs.setValue(kvsKeyName, fs.readFileSync(file), { contentType: 'video/mp4' });
+        keyValueStoreLink = kvs.getPublicUrl(kvsKeyName);
+    }
+    const transformedResult = {
+        testSuiteTitle: result?.runs[0].tests[0].title[0],
+        totalPassed: result?.totalPassed,
+        totalPending: result?.totalPending,
+        totalFailed: result?.totalFailed,
+        totalSkipped: result?.totalSkipped,
+        totalDuration: result?.totalDuration,
+        videoLink: keyValueStoreLink,
+        rawData: result,
+    };
+    await dataset.pushData(transformedResult);
+}
+
```

Comment:
It'd be cool to also log the result of the tests here. Something like:

```
Results:

10 of 10 passed (100%)
 
```

Reply:
Maybe even what tests passed/failed so you can see it in the log without coming to the kv store => better UX

Reply:
Great idea, I have added the Summary table at the end of the log

<img width="505" alt="Screenshot 2023-03-15 at 15 08 10" src="https://user-images.githubusercontent.com/56041262/225334056-992559c2-5b67-4a49-907e-48548c040262.png">

Reply:
This is cool! 👍